pallets/pools/src/lib.rs

```diff
@@ -838,38 +890,75 @@ pub mod pallet {
 					})
 					.collect::<Vec<_>>();
 
-				let tranche_values = pool
+				let tranche_supply = pool
 					.tranches
 					.iter()
 					.map(|tranche| tranche.debt.checked_add(&tranche.reserve))
 					.collect::<Option<Vec<_>>>()
 					.ok_or(Error::<T>::Overflow)?;
 
+				let seniorities = pool
+					.tranches
+					.iter()
+					.map(|tranche| tranche.seniority)
+					.collect::<Vec<_>>();
+
+				let tranche_min_risk_buffs = pool
+					.tranches
+					.iter()
+					.map(|tranche| tranche.min_risk_buffer)
+					.collect::<Vec<_>>();
+
 				let epoch_tranches: Vec<_> = orders
 					.iter()
-					.zip(&tranche_values)
+					.zip(&tranche_supply)
 					.zip(&epoch_tranche_prices)
-					.map(|(((invest, redeem), value), price)| EpochExecutionTranche {
-						value: *value,
-						price: *price,
-						invest: *invest,
-						redeem: *redeem,
-					})
+					.zip(&seniorities)
+					.zip(&tranche_min_risk_buffs)
+					.map(
```

Comment:
Now that we're passing in several fields of the tranche, I'm not sure it makes sense to precollect things into vecs?. We should be able to just pass in the tranche (untested, written in github comment):


```
let epoch_tranches = orders
    .iter()
    .zip(&tranches)
    .zip(&epoch_tranche_prices)
    .map(|(i((nvest, redeem), tranche), price) {
        let supply = tranche.debt.checked_add(&tranche.reserve)?;
        Some(EpochExecutionTranche {
		supply,
		price: *price,
		invest: *invest,
		redeem: *redeem,
		seniority: tranche.seniority,
		min_risk_buffer: tranche.min_risk_buffer,
	})
    })
    .collect::<Option<Vec<_>>>()
    .ok_or(Error::<T>::Overflow)?;
```

Reply:
I am on your side here. But I would keep this refactoring for another PR in order to keep the scope as is. I would suggest we do this when we decide on a refactor for the pool and how we handle the tranches.